overlord/snapstate/flags.go

```diff
@@ -50,3 +54,32 @@ func (f Flags) ForSnapSetup() Flags {
 	f.IgnoreValidation = false
 	return f
 }
+
+// EffectiveConfinement return the effective confinement for a given declared
+// confinement and flags like jailmode or devmode.
+func (f Flags) EffectiveConfinement(confinementType snap.ConfinementType) snap.ConfinementType {
+	switch confinementType {
+	case snap.ClassicConfinement:
+		// unchanged, flags don't impact snaps with classic confinement
```

Comment:
We discussed this online and agreed that the proper way to go is likely with a type holding boolean flags, similar to overlord/snapstate/flags.go.